data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/RecordDispatcher.java

```diff
@@ -128,6 +133,21 @@ public void handle(final KafkaConsumerRecord<String, CloudEvent> record) {
 
   private void onRecordReceived(final KafkaConsumerRecord<String, CloudEvent> record, Promise<Void> finalProm) {
     logDebug("Handling record", record);
+
+    // Trace record received event
+    if (consumerTracer != null) {
+      Context context = Vertx.currentContext();
+      ConsumerTracer.StartedSpan span = consumerTracer.prepareMessageReceived(context, record.record());
+      finalProm.future()
+        .onComplete(ar -> {
+          if (ar.succeeded()) {
+            span.finish(context);
+          } else {
+            span.fail(context, ar.cause());
+          }
+        });
```

Comment:
Could span be null? When propagation policy is like propagate but there is no upstream span?

Reply:
yep it could be, i fixed the code